components/Discussion/CommentContainers/CommentContainer.tsx

```diff
@@ -0,0 +1,141 @@
+import React, { ReactElement, useCallback, useMemo, useState } from 'react'
+import { CommentNode, CommentProps } from '@project-r/styleguide'
+import { useTranslation } from '../../../lib/withT'
+import Link from 'next/link'
+import { getFocusHref } from '../shared/CommentLink'
+import { format } from 'url'
+import { useDiscussion } from '../context/DiscussionContext'
+import useVoteCommentHandlers from '../hooks/actions/useVoteCommentHandlers'
+import { CommentTreeNode } from '../helpers/makeCommentTree'
+import getCommentMenuItems from './getCommentActions'
+import { useMe } from '../../../lib/context/MeContext'
+import useReportCommentHandler from '../hooks/actions/useReportCommentHandler'
+import useUnpublishCommentHandler from '../hooks/actions/useUnpublishCommentHandler'
+import DiscussionComposer from '../DiscussionComposer/DiscussionComposer'
+
+type Props = {
+  CommentComponent?: React.ElementType<CommentProps>
+  comment: CommentTreeNode
+  isLast?: boolean
+}
+
+const CommentContainer = ({
+  CommentComponent = CommentNode,
+  comment,
+  isLast
+}: Props): ReactElement => {
+  const [isEditing, setIsEditing] = useState(false)
+  const [isReplying, setIsReplying] = useState(false)
+
+  const { t } = useTranslation()
+  const { me } = useMe()
+  const {
+    id: discussionId,
+    discussion,
+    fetchMore,
+    overlays: { shareOverlay, featureOverlay }
+  } = useDiscussion()
+
+  // Handlers
+  const voteHandlers = useVoteCommentHandlers()
+  const reportCommentHandler = useReportCommentHandler()
+  const unpublishCommentHandler = useUnpublishCommentHandler()
+
+  const menuItems = useMemo(
+    () =>
+      getCommentMenuItems({
+        t,
+        comment,
+        setEditMode: setIsEditing,
+        roles: me?.roles ?? [],
+        actions: {
+          reportCommentHandler,
+          unpublishCommentHandler,
+          featureCommentHandler: featureOverlay.handleOpen
+        }
+      }),
+    [
+      t,
+      comment,
+      setIsEditing,
+      me?.roles,
+      reportCommentHandler,
+      unpublishCommentHandler,
+      featureOverlay.handleOpen
+    ]
+  )
+
+  const focusHref = useMemo(() => {
+    const urlObject = getFocusHref(discussion, comment)
+    return format(urlObject)
+  }, [discussion, comment])
+
+  const profileHref = useMemo(
```

Comment:
A memo is totally unnecessary for a simple if statement – the memo costs more than the ternary expression.